packages/orchestration/test/examples/stake-ica.contract.test.ts

```diff
@@ -0,0 +1,156 @@
+
+const getChainTerms = (chainName: keyof typeof chainInfo): StakeIcaTerms => {
+  const { chainId, stakingTokens, icqEnabled } = chainInfo[chainName];
+  const agoricConns = chainInfo.agoric.connections;
+  return {
+    chainId,
+    hostConnectionId: agoricConns[chainId].counterparty.connection_id,
+    controllerConnectionId: agoricConns[chainId].id,
+    bondDenom: stakingTokens[0].denom,
+    icqEnabled,
+  };
+};
+
+const startContract = async ({
+  orchestration,
+  timer,
+  marshaller,
+  storage,
+  issuerKeywordRecord = undefined,
+  terms = getChainTerms('cosmoshub'),
+  storagePath = 'stakeAtom',
+}) => {
+  const { zoe, bundleAndInstall } = await setUpZoeForTest();
+  const installation: Installation<StartFn> =
+    await bundleAndInstall(contractFile);
+
+  const { publicFacet } = await E(zoe).startInstance(
+    installation,
+    issuerKeywordRecord,
+    terms,
+    {
+      marshaller,
+      orchestration,
+      storageNode: storage.rootNode.makeChildNode(storagePath),
+      timer,
+    },
+  );
+  return { publicFacet, zoe };
+};
+
+test('makeAccount, getAddress, getBalances, getBalance', async t => {
+  const { bootstrap } = await commonSetup(t);
+  {
+    // stakeAtom
+    const { publicFacet } = await startContract(bootstrap);
+
+    t.log('make an ICA account');
+    const account = await E(publicFacet).makeAccount();
+    t.truthy(account, 'account is returned');
+    const chainAddress = await E(account).getAddress();
+    t.regex(chainAddress.address, /cosmos1/);
+    t.like(chainAddress, { chainId: 'cosmoshub-4', addressEncoding: 'bech32' });
+
+    await t.throwsAsync(E(account).getBalances(), {
+      message: 'not yet implemented',
+    });
+
+    await t.throwsAsync(E(account).getBalance('uatom'), {
+      message: 'Queries not available for chain "cosmoshub-4"',
+    });
+
+    const accountP = E(publicFacet).makeAccount();
+    const { address: address2 } = await E(accountP).getAddress();
+    t.regex(address2, /cosmos1/);
+    t.not(chainAddress.address, address2, 'account addresses are unique');
+  }
+  {
+    // stakeOsmo
+    const { ibcBridge } = bootstrap;
+    await E(ibcBridge).setAddressPrefix('osmo');
+    const { publicFacet } = await startContract({
+      ...bootstrap,
+      terms: getChainTerms('osmosis'),
+      storagePath: 'stakeOsmo',
+    });
+
+    const account = await E(publicFacet).makeAccount();
+    t.truthy(account, 'account is returned');
+    const chainAddress = await E(account).getAddress();
+    t.regex(chainAddress.address, /osmo1/);
+    t.like(chainAddress, { chainId: 'osmosis-1' });
+
+    await E(ibcBridge).setMockAck({
+      [buildQueryPacketString([
+        QueryBalanceRequest.toProtoMsg({
+          address: chainAddress.address,
+          denom: 'uosmo',
+        }),
+      ])]: buildQueryResponseString(QueryBalanceResponse, {
+        balance: { amount: '0', denom: 'uosmo' },
+      }),
+    });
```

Comment:
this is hard to parse visually. consider pulling out the key and maybe the value into named bindings